cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/replication/PullStrategy.java

```diff
@@ -100,18 +102,23 @@
     public PullStrategy(URI source,
                         Database target,
                         PullFilter filter,
+                        PullSelector selector,
                         List<HttpConnectionRequestInterceptor> requestInterceptors,
                         List<HttpConnectionResponseInterceptor> responseInterceptors) {
         this.filter = filter;
+        this.selector = selector;
         this.sourceDb = new CouchClientWrapper(new CouchClient(source, requestInterceptors,
                 responseInterceptors));
         this.targetDb = new DatastoreWrapper((DatabaseImpl) target);
         String replicatorName;
-        if (filter == null) {
-            replicatorName = String.format("%s <-- %s ", target.getPath(), source);
-        } else {
+        if (filter != null) {
             replicatorName = String.format("%s <-- %s (%s)", target.getPath(), source,
                     filter.getName());
+        } else if (selector != null) {
```

Comment:
The current logic means that if a filter and a selector are both set, the filter takes precedence. I don't think this makes sense and it's not obvious to the user.

Maybe it would be better to throw an exception either here or in the builder if both are set.

Reply:
Ok, I'll do it in the builder